lib/phoenix_html/form_data.ex

```diff
@@ -52,11 +52,14 @@ defimpl Phoenix.HTML.FormData, for: [Plug.Conn, Atom] do
   def to_form(conn_or_atom, opts) do
     {name, params, opts} = name_params_and_opts(conn_or_atom, opts)
     {errors, opts} = Keyword.pop(opts, :errors, [])
+    {id, opts} = Keyword.pop(opts, :id)
+
+    id = if is_nil(id), do: name, else: to_string(id) <> "_#{name}"
```

Comment:
My suggestion is to use `id || name` here too. :)

Reply:
Ow, that makes sense. Somehow I got confused thinking the name has been prefixed along with id 😅